### ISSUE TITLE
TextFieldHint ID does not use user provided ID

### ISSUE THREAD

**Issue by davidjacob93** [manchesergit/material-ui#238]
Added unit tests to #204. User providing ID test fails due to the appending of string to the ID provided. Changing the code at component level breaks the accessibility tests.

![capture9](https://user-images.githubusercontent.com/25061353/28921626-9730ea84-784e-11e7-95a9-f2d7f1de5fc3.PNG)
